src/main/kotlin/org/rust/ide/template/macros/utils.kt

```diff
@@ -0,0 +1,15 @@
+package org.rust.ide.template.macros
+
+import com.intellij.psi.PsiElement
+import org.rust.lang.core.psi.RustCompositeElement
+import org.rust.lang.core.psi.RustPatBindingElement
+import org.rust.lang.core.resolve.RustResolveEngine
+
+fun getPatBindingsVisibleAt(place: PsiElement?): List<RustPatBindingElement> {
+    if (place == null) return emptyList()
+    val pivot = place as? RustCompositeElement
+    return RustResolveEngine.enumerateScopesFor(place)
+        .flatMap { RustResolveEngine.declarations(it, pivot) }
+        .mapNotNull { it as? RustPatBindingElement }
+        .toList()
```

Comment:
Ok, so this has two issues which need to be fixed before merging:
 - The requirement for pivot to be `RustCompositeElement` cannot be fulfilled in most situations, because place is mostly `PsiWhitespace`. Can this limitation be removed?
 - This is *incredibly* slow